saleor/payment/tasks.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from datetime import datetime
+
+import pytz
+from django.conf import settings
+from django.db.models import OuterRef, Q, Subquery
+
+from ..celeryconf import app
+from ..channel.models import Channel
+from ..checkout import CheckoutAuthorizeStatus, CheckoutChargeStatus
+from ..checkout.models import Checkout
+from ..payment.models import TransactionEvent, TransactionItem
+from ..plugins.manager import get_plugins_manager
+from . import PaymentError, TransactionEventType
+from .gateway import request_cancelation_action, request_refund_action
+
+logger = logging.getLogger(__name__)
+
+CHECKOUT_BATCH_SIZE = 30
+TRANSACTION_BATCH_SIZE = 60
+
+
+def checkouts_with_funds_to_release():
+    now = datetime.now(pytz.UTC)
+
+    return Checkout.objects.filter(
+        (
+            Q(automatically_refundable=True)
+            & Q(last_change__lt=now - settings.CHECKOUT_TTL_BEFORE_RELEASING_FUNDS)
+            & Q(
+                last_transaction_modified_at__lt=now
+                - settings.CHECKOUT_TTL_BEFORE_RELEASING_FUNDS
+            )
+        )
+        & (
+            ~Q(authorize_status=CheckoutAuthorizeStatus.NONE)
+            | ~Q(charge_status=CheckoutChargeStatus.NONE)
+        )
+    )
+
+
+@app.task
+def transaction_release_funds_for_checkout_task():
+    checkouts = checkouts_with_funds_to_release().order_by("last_change")
+    checkout_pks = checkouts.values_list("pk", flat=True)[:CHECKOUT_BATCH_SIZE]
+    if checkout_pks:
+        transaction_events = TransactionEvent.objects.filter(
+            transaction_id=OuterRef("pk"),
+        ).order_by("-created_at")
+
+        checkout_subquery = Checkout.objects.filter(
+            pk=OuterRef("checkout_id")
+        ).annotate(
+            channel_slug=Subquery(
+                Channel.objects.filter(pk=OuterRef("channel_id")).values("slug")[:1]
+            )
+        )
```

Comment:
Maybe we could generate `checkout_pk_to_channel_slug_map` instead of generating a complicated subquery with many annotations.